nextflow_schema.json

```diff
@@ -479,6 +479,82 @@
+                    "help_text": "Specify a mapping quality threshold for mapped reads to be kept for downstream analysis.\n\nBy default all reads are retained and is therefore set to 0 to ensure no quality filtering is performed.\n\nNote that by default the output BAM files of this step are _not_ stored in the results directory (as it is assumed that deduplicated BAM files are preferred). See `--bamfiltering_savefilteredbams` if you wish to save these.\n\n> Modifies tool parameter(s):\n> - samtools view `-q`",
+                    "fa_icon": "fas fa-thermometer-full"
+                },
+                "bamfiltering_generateunmappedfastq": {
+                    "type": "boolean",
+                    "description": "Generate FASTQ files containing only unmapped reads from the aligner generated BAM files.",
+                    "help_text": "This turns on the generation and saving in the results directory of FASTQs of only the unmapped reads from the mapping step using `samtools fastq`.\n\nThis could be useful if you wish to do other analysis of the unmapped reads independently of the pipeline.\n\nNote the reads in these FASTQ files have _not_ undergone length of quality filtering\n\n> Modifies tool parameter(s):\n> - samtools fastq: `-f 4`",
+                    "fa_icon": "fas fa-file-alt"
+                },
+                "bamfiltering_generatemappedfastq": {
+                    "type": "boolean",
+                    "description": "Generate FASTQ files containing only mapped reads from the aligner generated BAM files .",
+                    "help_text": "This turns on the generation and saving in the results directory of FASTQs of only the mapped reads from the mapping step, using `samtools fastq`.\n\nThis could be useful if you wish to do other analysis of the mapped reads independently of the pipeline, such as remapping with different parameters (whereby only including mapped reads will speed up computation time during the re-mapping due to reduced input data).\n\nNote the reads in these FASTQ files have _not_ undergone length of quality filtering\n\n> Modifies tool parameter(s):\n> - samtools fastq: `-F 4`",
+                    "fa_icon": "far fa-file-alt"
+                },
+                "bamfiltering_retainunmappedgenomicbam": {
+                    "type": "boolean",
+                    "description": "Specify to retain unmapped reads in the BAM file used for downstream genomic analyses.",
+                    "help_text": "You can use this parameter to retain unmapped reads (optionally also length filtered) in the genomic BAM for downstream analysis - as by default, the pipeine only keeps mapped reads for downstream.\n\nThis is also turned on if `--metagenomicscreening_input` is set to all.\n\n> ⚠️ This will likely slow down run time of downstream pipeline steps!\n\n> Modifys tool parameter(s):\n> - samtools view: `-f 4` / `-F 4`",
```

Comment:
```suggestion
                    "help_text": "You can use this parameter to retain unmapped reads (optionally also length filtered) in the genomic BAM for downstream analysis. By default, the pipeline only keeps mapped reads for downstream analysis.\n\nThis is also turned on if `--metagenomicscreening_input` is set to `all`.\n\n> ⚠️ This will likely slow down run time of downstream pipeline steps!\n\n> Modifies tool parameter(s):\n> - samtools view: `-f 4` / `-F 4`",
```

Reply:
This also only works if you then DON'T deduplicate, since that will inherently filter for mapped reads only atm